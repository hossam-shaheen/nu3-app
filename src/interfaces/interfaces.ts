export interface ProductType {
    id: number;
    title: string;
    price: number;
    image: string;
    vendor: string;
    link: string;
}

export interface ProductsType {
    totalItems: number;
    items: ProductType[];
}

export interface ProductsSearchProps {
    onSearch: (searchKeyWord: string) => void;
}

export interface SortProps {
    onSort: (sortKeyWord: string) => void;
}
export interface ImageType {
    id: number;
    src: string;
    alt: string;
}
export interface ProductAttributeProps {
    productImage: ImageType;
    productImages: ImageType[];
    ProductTitle: string;
    productBodyHtml: string;
    productVendor: string;
}


export interface ProductVariantsProps {
    options: OptionsType[];
    variants: VariantType[];
}


export interface ProductInfoType {
    id: number;
    title: string;
    body_html: string;
    vendor: string;
    images: ImageType[];
    image: ImageType;
    options: OptionsType[];
    variants: VariantType[];
    url: string;
}

export interface VariantType {
    id: number,
    title: string,
    price: number,
}

export interface OptionsType {
    id: number,
    values: string[],
    name: string,
}

export interface ErrorType {
    icon?: string,
    message: string
}