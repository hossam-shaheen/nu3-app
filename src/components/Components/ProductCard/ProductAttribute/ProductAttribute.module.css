.product-attribute {
    display: flex;
    justify-content: space-between;
    width: calc(100% - 155px);
}

.product-title {
    font-size: 35px;
    font-weight: 500;
    margin-top: 0;
    margin-bottom: 20px;
}

.product-vendor {
    color: var(--secondary);
    margin-top: 0;
    margin-bottom: 20px;
    font-weight: 600;
}

.product-vendor span:first-child {
    color: #000;
    margin-right: 5px;
    margin-bottom: 20px;
    font-weight: 400;
    font-size: 14px;
}

.product-html li {
    font-size: 14px;
    margin-bottom: 10px;
}