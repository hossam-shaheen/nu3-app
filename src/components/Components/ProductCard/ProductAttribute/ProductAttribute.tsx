import { FunctionComponent } from "react";
import { ProductAttributeProps } from "src/interfaces/interfaces";
import classes from "./ProductAttribute.module.css";

export const ProductAttribute: FunctionComponent<{
  productAttributes: ProductAttributeProps | null;
}> = ({ productAttributes }): JSX.Element => {
  const productBodyHtml = productAttributes?.productBodyHtml.split("✓") || [];

  return (
    <div className={classes["product-attribute"]}>
      <div className={classes["product-info"]}>
        {productAttributes?.ProductTitle && (
          <h1 className={classes["product-title"]}>
            {productAttributes.ProductTitle}
          </h1>
        )}
        {productAttributes?.productVendor && (
          <p className={classes["product-vendor"]}>
            <span>Von</span>
            <span>{productAttributes.productVendor}</span>
          </p>
        )}
        {productBodyHtml.length > 0 && (
          <ul className={classes["product-html"]}>
            {productBodyHtml?.map((bodyHtml, index) => {
              return <li key={index}>✓ {bodyHtml}</li>;
            })}
          </ul>
        )}
      </div>
    </div>
  );
};
export default ProductAttribute;
