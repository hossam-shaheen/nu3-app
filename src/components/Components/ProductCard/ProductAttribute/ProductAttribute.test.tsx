import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ProductAttributeProps } from 'src/interfaces/interfaces';
import ProductAttribute from './ProductAttribute';

const mockedProductAttributes: ProductAttributeProps = {
    productImage: {
        id: 1,
        src: "https://cdn.shopify.com/s/files/1/0289/1534/products/product-image-large.jpg?v=1571608031",
        alt: "Product Image"
    },
    productImages: [{
        id: 1,
        src: "https://cdn.shopify.com/s/files/1/0289/1534/products/product-image-large.jpg?v=1571608031",
        alt: "Product Image"
    }],
    ProductTitle: "nu3 Bio Acerola",
    productBodyHtml: "nu3 Bio Acerola-Pulver: 100 g ✓ Natürliche Vitamin C-Quelle ✓ Ohne Zusatzstoffe wie Maltodextrin ✓ Schonend gefriergetrocknet. Jetzt im nu3 Shop bestellen!",
    productVendor: "nu3",
};


test('ProductAttribute pass ProductAttributes Props : Renders', () => {
    render(<ProductAttribute
        productAttributes={mockedProductAttributes}
    />);

    const productTitle = screen.getByRole("heading");
    expect(productTitle).toBeInTheDocument();

    const productVendor = screen.getByText("nu3");
    expect(productVendor).toHaveTextContent("nu3");

    const productHtmlList = screen.getAllByRole("listitem");
    expect(productHtmlList.length).toBe(4);

});


test(`ProductAttribute didn't pass ProductAttributes Props : Renders`, () => {
    render(<ProductAttribute
        productAttributes={null}
    />);

    const productTitle = screen.queryByRole("heading");
    expect(productTitle).not.toBeInTheDocument();

    const productVendor = screen.queryByText("nu3");
    expect(productVendor).not.toBeInTheDocument();

    const productHtmlList = screen.queryByRole("listitem");
    expect(productHtmlList).not.toBeInTheDocument();

});


test(`ProductAttribute pass ProductAttributes Props: Snapshot`, () => {
    const component = renderer.create(<ProductAttribute
        productAttributes={mockedProductAttributes}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test(`ProductAttribute didn't pass ProductAttributes Props : Snapshot`, () => {
    const component = renderer.create(<ProductAttribute
        productAttributes={null}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

