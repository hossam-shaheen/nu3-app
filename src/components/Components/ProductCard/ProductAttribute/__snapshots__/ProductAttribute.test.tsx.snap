// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductAttribute didn't pass ProductAttributes Props : Snapshot 1`] = `
<div
  className="product-attribute"
>
  <div
    className="product-info"
  />
</div>
`;

exports[`ProductAttribute pass ProductAttributes Props: Snapshot 1`] = `
<div
  className="product-attribute"
>
  <div
    className="product-info"
  >
    <h1
      className="product-title"
    >
      nu3 Bio Acerola
    </h1>
    <p
      className="product-vendor"
    >
      <span>
        Von
      </span>
      <span>
        nu3
      </span>
    </p>
    <ul
      className="product-html"
    >
      <li>
        ✓ 
        nu3 Bio Acerola-Pulver: 100 g 
      </li>
      <li>
        ✓ 
         Natürliche Vitamin C-Quelle 
      </li>
      <li>
        ✓ 
         Ohne Zusatzstoffe wie Maltodextrin 
      </li>
      <li>
        ✓ 
         Schonend gefriergetrocknet. Jetzt im nu3 Shop bestellen!
      </li>
    </ul>
  </div>
</div>
`;
